final_task/rss_reader/validation_functions.py

```diff
@@ -0,0 +1,82 @@
+""" Module of validation functions.
+
+    Functions:
+    check_url(com_line_args, logger) -> True
+    check_internet_connection(com_line_args, logger) -> True
+    check_emptiness(news_collection, logger) -> True
+    check_version_arg(com_line_args, logger) -> None
+    check_limit_arg(news_collection, com_line_args, logger) -> limit (int) """
+
+import requests
+from urllib.request import Request, urlopen
+from urllib.error import URLError
+
+
+def check_url(com_line_args, logger):
+    """ Check URL function. """
+    req = Request(com_line_args.source)
+    try:
+        logger.info("Checking url.")
+        response = urlopen(req)
+    except URLError as e:
+        if hasattr(e, "reason"):
+            logger.error(f"Failed to reach a server. ")
+            print("Please, check your URL.")
+            exit()
```

Comment:
It is better to have only one exit point from the application. For example, you can raise your custom exception here and handle it in the main function

Reply:
fixed